### PR TITLE
Added Java command line client

### PR DESCRIPTION
**Pull request by meteatamel**

### CODE REVIEW

**Comment by thesandlord** [comment]
LGTM, @markmandel can you take a look and merge?

File: client/java-cli/src/main/java/io/grpc/examples/simonsays/Game.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    public Game(ConsoleWindow window,
+            SimonSaysGrpc.SimonSaysStub asyncStub) {
+        this.window = window;
+        this.asyncStub = asyncStub;
+
+        setupStreamObserver();
+    }
+
+    /**
```
Comment (markmandel) [inline client/java-cli/src/main/java/io/grpc/examples/simonsays/Game.java:52]: Should this comment still be here?

File: client/java-cli/src/main/java/io/grpc/examples/simonsays/Game.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+     * STEP4c: Return the startedGame boolean.
+     */
+    public boolean isStarted() {
+        return started;
+    }
+
+    /**
+     * STEP3: Setup stream observer. In order to setup stream observer, you need:
+     * 
+     * STEP3a: Implement handleResponse.
```
Comment (markmandel) [inline client/java-cli/src/main/java/io/grpc/examples/simonsays/Game.java]: Should these Implement comments still be here?

**Comment by markmandel** [comment]
A couple of minor comment tweaks, but otherwise, LGTM. Good to merge one the comments are fixed.

**Comment by meteatamel** [comment]
The comments are useful for the workshop...We have the same in node.js client.

**Comment by markmandel** [comment]
In that case. I say, good to merge.

**Comment by markmandel** [comment]
Actually.. let me come back on that again. If this is a completed implementation, why should the comments be there? People don't know this is specifically for a workshop.

**Comment by markmandel** [comment]
My next nit picky issue, is that the Javadocs are inconsistently applied.

This is particularly true in ConsoleInputReader (and when reading the code, it took me a while to work out what the functions where doing) - let's add Javadocs to everything

File: client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    /**
+     * Read a line for a question with a default value.
+     */
+    private String readLine(String question, Object defaultValue) {
+        System.out.print(getFormattedQuestion(question, defaultValue));
+        String line =  scanner.nextLine();
+        return line;
+    }
+
+    private static String getFormattedQuestion(String question, Object defaultValue) {
```
Comment (markmandel) [inline client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java:105]: Javadoc please :)

File: client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        return String.format(INPUT_FORMAT, question, defaultValue);
+    }
+
+    private static boolean isNullOrEmpty(String str) {
```
Comment (markmandel) [inline client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java:112]: Javadoc please :)

File: client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        return str == null || str.isEmpty();
+    }
+
+    private static Color doReadColor(String line) {
```
Comment (markmandel) [inline client/java-cli/src/main/java/io/grpc/examples/simonsays/ConsoleInputReader.java:119]: Javadoc please :)

**Comment by markmandel** [comment]
LGTM!

**Pull request merged**